src/components/Articles.jsx

```diff
@@ -5,29 +5,44 @@ import NavArticle from "./NavArticle";
 import ListArticles from "./ListAritcles.jsx";
 
 const Articles = () => {
-
+const { topic } = useParams();
 const [isLoading, setIsLoading] = useState(true);
 const [articles, setArticles] = useState([])
-
-const { topic } = useParams()
+const [params, setParams ] = useState({})
 
 
 useEffect(() => {
     setIsLoading(true);
-    getArticles(topic).then((res) => {
-        setArticles(res)
+    getArticles(params).then(({ articles }) => {
+        setArticles(articles)
         setIsLoading(false);
     })
+}, [params])
+
+useEffect(() => {
+    setParams({ topic })
 }, [topic])
 
+const handleSort = (column) => {
+    setParams((currParams) => {
+        return { ...currParams, sort_by: column}
+    })
+}
+
 if(isLoading) return <p>Loading...</p>
 
 return (
     <div>
         <div>
             <NavArticle />
+            <select>
+                <option onClick={() => handleSort('title')}value='title'>Title</option>
```

Comment:
`<option>` tags do not have an onClick. You need to put an `onChange` in the `<select>` and use `event.target.value`, as detailed above.